framework/quota/src/main/java/org/apache/cloudstack/quota/QuotaAlertManagerImpl.java

```diff
@@ -325,14 +326,20 @@ public void sentSuccessfully(final QuotaAccountDao quotaAccountDao) {
         }
     };
 
-    protected void sendQuotaAlert(List<String> emails, String subject, String body) {
+    protected void sendQuotaAlert(String accountUuid, List<String> emails, String subject, String body) {
         SMTPMailProperties mailProperties = new SMTPMailProperties();
 
         mailProperties.setSender(new MailAddress(senderAddress));
         mailProperties.setSubject(subject);
         mailProperties.setContent(body);
         mailProperties.setContentType("text/html; charset=utf-8");
 
+        if (CollectionUtils.isEmpty(emails)) {
+            s_logger.warn(String.format("Unable to send quota alert email with subject [%s] and content [%s]. "
```

Comment:
Not a big deal but `Unable to send quota alert email` might give different meaning like something wrong with email system or configuration but the actual reason is the empty list. Just mentioning that the list is empty should be good enough

Reply:
Thanks for the review @ravening
I changed the message to make it more evident that quota alert emails are not being sent just because the email list is empty. But I found it important to keep the "subject" and "content" information, because it makes it easier to decide if it's something crucial to deal with or not.